contracts/ServiceNodeContribution.sol

```diff
@@ -134,16 +134,33 @@ contract ServiceNodeContribution is Shared {
     }
 
     /**
-     * @notice Function to reset the contract to an empty state
+     * @notice Reset the contract and refund the contributions to the operator
+     * and contributors. The operator must re-initialise the contract
      * @param amount The amount of funds the operator is to contribute.
      */
     function resetContract(uint256 amount) external onlyOperator {
         require(finalized, "Contract has not been finalized yet.");
-        finalized = false;
-        operatorContribution = 0;
-        totalContribution = 0;
-        numberContributors = 0;
+
+        // NOTE: Remove and refund all contributors (including the operator)
+        for (uint256 index = 0; index < numberContributors; index++) {
+            // NOTE: We return the contributions in reverse order (due to
+            // swap-and-pop pattern) for more predictability in return sequence.
+            uint256 reverseIndex = numberContributors - (index + 1);
+            address toRemove     = contributorAddresses[reverseIndex];
+            removeAndRefundContributor(toRemove);
```

Comment:
There should be zero funds in the contract at this point. The contract is finalised, sends all its funds to the StakingRewardsContract and this contract is now empty.

Reset Contract is then called if you want to reuse this contract, if your node gets withdrawn and you want to save yourself deploying another contribution contract to get it back up

Reply:
That's true yes. I had updated this after I learnt of this and fixed it in https://github.com/oxen-io/eth-sn-contracts/pull/21/commits/eb5580995d267572ab646a6608035229a5af5b77